src/librados/librados_cxx.cc

```diff
@@ -266,6 +268,20 @@ void librados::ObjectReadOperation::omap_get_vals2(
   std::map<std::string, bufferlist> *out_vals,
   bool *pmore,
   int *prval)
+{
+  omap_get_vals2(
+    start_after, filter_prefix, max_return,
+    reinterpret_cast<std::map<std::string, bufferlist, std::less<>>*>(out_vals),
```

Comment:
It's hard to expect that `std::less<std::string>` will differ in terms of memory layout from `std::less<>`. Maybe, for the sake of paranoia, we could verify e.g. sizes with compile-time assertions?